connect/runtime/src/main/java/org/apache/kafka/connect/runtime/distributed/DistributedHerder.java

```diff
@@ -27,11 +27,13 @@
 import org.apache.kafka.connect.errors.NotFoundException;
 import org.apache.kafka.connect.runtime.AbstractHerder;
 import org.apache.kafka.connect.runtime.ConnectorConfig;
+import org.apache.kafka.connect.runtime.ConnectorStatus;
```

Comment:
Not a big fan of using magic constants/variables etc as parameters, even if that's a boolean. 

Here's your article of the day on the [matter](http://www.drdobbs.com/conversationstruth-or-consequences/184403845). 

Enums are at least more self-documenting. But I was wary too in reusing an existing enum. Based on our discussion above, I'll probably define a private one as it's done in other places too, if by the end of this PR we still need one as a parameter. 

P.S. please use the term "race condition" with caution. Raises particularly alarming red flags, especially in the early morning :)